lisa/lisa-core/src/main/java/it/unive/lisa/analysis/string/Prefix.java

```diff
@@ -0,0 +1,199 @@
+package it.unive.lisa.analysis.string;
+
+import it.unive.lisa.analysis.Lattice;
+import it.unive.lisa.analysis.SemanticDomain.Satisfiability;
+import it.unive.lisa.analysis.SemanticException;
+import it.unive.lisa.analysis.nonrelational.value.BaseNonRelationalValueDomain;
+import it.unive.lisa.analysis.representation.DomainRepresentation;
+import it.unive.lisa.analysis.representation.StringRepresentation;
+import it.unive.lisa.program.cfg.ProgramPoint;
+import it.unive.lisa.symbolic.value.Constant;
+import it.unive.lisa.symbolic.value.operator.binary.*;
+import it.unive.lisa.symbolic.value.operator.ternary.TernaryOperator;
+import it.unive.lisa.symbolic.value.operator.unary.UnaryOperator;
+import java.util.Objects;
+
+/**
+ * The prefix string abstract domain.
+ *
+ * @author <a href="mailto:vincenzo.arceri@unipr.it">Vincenzo Arceri</a>
+ * @author <a href="mailto:sergiosalvatore.evola@studenti.unipr.it">Sergio
+ *             Salvatore Evola</a>
+ * 
+ * @see <a href=
+ *          "https://link.springer.com/chapter/10.1007/978-3-642-24559-6_34">
+ *          https://link.springer.com/chapter/10.1007/978-3-642-24559-6_34</a>
+ */
+public class Prefix extends BaseNonRelationalValueDomain<Prefix> {
+
+	private final static Prefix TOP = new Prefix();
+	private final static Prefix BOTTOM = new Prefix(null);
+	private final String prefix;
+
+	/**
+	 * Builds the top prefix abstract element.
+	 */
+	public Prefix() {
+		this("");
+	}
+
+	/**
+	 * Builds a prefix abstract element.
+	 * 
+	 * @param prefix the prefix
+	 */
+	public Prefix(String prefix) {
+		this.prefix = prefix;
+	}
+
+	@Override
+	public Prefix lubAux(Prefix other) throws SemanticException {
+		String otherPrefixString = other.getPrefix();
```

Comment:
Valid here and in the other classes: as those getters just return the field, it's faster to access the field directly instead of calling the method